### PR TITLE
fix: violate hooks rule close gh-3108

### LINKED ISSUE #3108: [Crash] Error: Minified React error #321
I was *doing something...*, then Mask report an error.

> Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.

Error stack:

<pre>Error: Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at Object.Wh (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:233:238)
    at exports.useMemo (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/content-script.js:141:450)
    at useObservableValues (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/4671.js:9947:43)
    at usePostInfoDetails (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/9839.js:937:16)
    at PostInspector (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/chunk.9881.js:23279:66)
    at Yh (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:234:137)
    at Hk (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:355:96)
    at Gk (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:344:22)
    at Dk (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:343:370)
    at vk (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:343:230)</pre>

## Build info
- Version: 1.32.0
- NODE_ENV: production
- target: chromium
- build: stable
- architecture: web
- firefoxVariant: false
- resolution: desktop
- BUILD_DATE: 2021-05-07T15:13:35.015Z
- VERSION: v1.29.12-304-g771aa1a1

## Git ():

771aa1a1 (released) on tag "v1.32.0"
git@github.com:DimensionDev/Maskbook.git

### PR DESCRIPTION
**Pull request by Jack-Works**
closes #3108

I don't know if it works.

### CODE REVIEW

**Comment by nuanyang233** [comment]
The type-coverage did not pass, look like same proble in #3061.

**Review by guanbinrui — DISMISSED** [review]

**Comment by Jack-Works** [comment]
> The type-coverage did not pass, look like same problem in #3061.

Yes... I think maybe we should disable type-coverage? 🤔

**Comment by nuanyang233** [comment]
> > The type-coverage did not pass, look like same problem in #3061.
> 
> Yes... I think maybe we should disable type-coverage? 🤔

I found that type-coverage fails when there are changes to the contents of the `shared`.

**Comment by guanbinrui** [comment]
Hi @Jack-Works please fix those linter errors.

**Comment by Jack-Works** [comment]
> Hi @Jack-Works, please fix those linter errors.

The failing test comes from `type-coverage`. I and @nuanyang233 cannot reproduce the failure result from our machine.

**Comment by Jack-Works** [comment]
I have disabled type-coverage CI check. I don't know if it's really a good way to work with but yeah `I and @nuanyang233 cannot reproduce the failure result from our machine.`

cc @guanbinrui @septs @nuanyang233 @zhouhanseng

File: .github/workflows/linters.yml [hunk 1]
```diff
@@ -56,8 +56,8 @@ jobs:
           key: ${{ runner.os }}-node-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: ${{ runner.os }}-node-
       - run: npx pnpm install
-      - run: npm run lint:typecoverage
-        working-directory: packages/maskbook
+      # - run: npm run lint:typecoverage
```
Comment (septs) [inline .github/workflows/linters.yml:59]: plz add reason description, thanks

**Comment by Jack-Works** [comment]
@guanbinrui let's pick this and try if this fix the bug

**Comment by Jack-Works** [comment]
@septs @guanbinrui

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**